lib/azure/access_token.rb

```diff
@@ -0,0 +1,42 @@
+module Azure
+  class AccessToken
+    def initialize
+      @url = Azure.config.google_cloud_credentials[:credential_source][:url]
+      @conn = Faraday.new do |b|
+        b.response :json
+        b.request :url_encoded
+      end
+    end
+
+    def call
+      azure_token_response = @conn.get(@url) do |req|
+        req.body = request_body
+      end
```

Comment:
This is a weird one. We're making a get request but we're sending a request body. This is how I was able to do it with Faraday.

By default, the second arg to GET requests is query params, and the second arg to POST request is request body. I have to yield a block param to set the request body on a GET request.